core/ir/aarch64/encode.c

```diff
@@ -198,6 +199,80 @@ const reg_id_t dr_reg_fixer[] = { REG_NULL,
 };
 /* clang-format on */
 
+/* Maps containing-registers to their DR_ISA_REGDEPS virtual-register.
+ * Same size as dr_reg_fixer[], keep them synched.
+ * Note: DR_REG_INVALID (== DR_REG_V1 == 1) has been excluded to avoid failures on opnd_t
+ * operations.
+ */
+const reg_id_t dr_reg_virtual[] = {
+    DR_REG_V0,   DR_REG_V0,   DR_REG_V2,   DR_REG_V3,   DR_REG_V4,   DR_REG_V5,
```

Comment:
dr_reg_virtual[] mirrors dr_reg_fixer[] but with virtual registers, it's not a sequence, there are duplicates (like for dr_reg_fixer[]).
Note that in this specific case DR_REG_V1 would not appear anywhere because 1 is the DR_REG_INVALID for AARCH64.

Reply:
I find this very confusing to have V0 be NULL: I don't think that has any use case?  See my suggestion below to have V0 start at 2 and to skip INVALID for all arches to end up with an understandable cross-arch enum set.

Reply:
The reason to exclude DR_REG_INVALID (which would have DR_REG_V1 as its corresponding virtual register) is to avoid having a real ISA register map to DR_REG_V1.
This would raise an "Invalid register" error in, for example, instr_same() when we compare the operands of 2 regdeps instructions.

Reply:
> I find this very confusing to have V0 be NULL: I don't think that has any use case? See my suggestion below to have V0 start at 2 and to skip INVALID for all arches to end up with an understandable cross-arch enum set.

Ok, will change to that, so all DR_REG_V are valid.

Reply:
Done.